src/dual.jl

```diff
@@ -775,6 +775,17 @@ function LinearAlgebra.eigen(A::SymTridiagonal{<:Dual{Tg,T,N}}) where {Tg,T<:Rea
     Eigen(λ,Dual{Tg}.(Q, tuple.(parts...)))
 end
 
+# StepRange - default to operate with LinRange to avoid TwicePrecision #
+#----------------------------------------------------------------------#
+
+Base.:*(x::Dual, r::StepRangeLen{<:Real,<:Base.TwicePrecision}) =
+    LinRange(x * first(r), x * last(r), length(r))
+
+Base.:/(r::StepRangeLen{<:Real,<:Base.TwicePrecision}, x::Dual) =
+    LinRange(first(r) / x, last(r) / x, length(r))
+
+Base.:(:)(start::T, step::Dual, stop::T) where {T<:Real} = range(start, step, stop)
```

Comment:
You're right, I'm thinking it should have been something similar in logic to
`= range(start, stop, Int((stop - start) / step), step)`
but it seems to create some problems, like you pointed out.

Reply:
Maybe that's simplest. 

The calculation of length from step is where Base has some serious black magic. At least for floating point ranges; for integers of course it's easy. It may be possible to catch the one case which errors later in the pipe... but integer Dual numbers don't seem worth much effort.

Reply:
Just a thought: is it impossible to import the method that Base uses for calculating the length and applying it to the values and partials?

Reply:
I removed the line for the (:) operator IN [3319d5e](https://github.com/JuliaDiff/ForwardDiff.jl/pull/618/commits/3319d5e8c2c5b09e6e792256663547f37f57483e), as I don't think it's worth leaving this unmerged because of that; as you suggested, that could looked at in the future.